app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
```
--- a/app/controllers/application_controller/filter.rb
+++ b/app/controllers/application_controller/filter.rb
@@ -157,6 +157,10 @@ module ApplicationController::Filter
     # Rebuild the pulldowns if opening the search box
     @edit[@expkey].prefill_val_types unless @edit[:adv_search_open]
 
+    unless @edit[:adv_search_open]
+      @edit[:search_type] = @edit[:expression][:selected][:typ] if @edit.fetch_path(:expression, :selected, :typ).present?
+    end
+
     render :update do |page|
       page << javascript_prologue
       if @edit[:adv_search_open] == true
```
I think this is not needed here. For me this causes wrong behavior of saving a filter. For example if we select filter which has type `default`, this code allows us to save a filter with type `default` and I think we shouldn't be able to create/edit default filters (see `adv_search_name_typed` method).